### PR TITLE
Prepend sqllogictest explain result with line number

### LINKED ISSUE #9443: Format plan with line number.
### Is your feature request related to a problem or challenge?

When I try to complete #8416 , I found it hard to review changes in plan in tpch sqllogictests.

I think the main reason is that CoalesceBatchesExec and RepartitionExec::RoundRobinBatch would make git diff aligns the result in unexcepted position. 

Takes q11.slt.part as example,  it aligns CoalesceBatchesExec and RepartitionExec::RoundRobinBatch but ignore the `--` changes. 

图片

If we remove CoalesceBatchesExec and RepartitionExec::RoundRobinBatch, it works well to find the `--` changes. 

图片

图片

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by duongcongtoai**
## Which issue does this PR close?

Closes #9443.

### CODE REVIEW

**Comment by my-vegetable-has-exploded** [comment]
I think ci maybe happy after add feature avro. 
I don't know whether `INCLUDE_TPCH=true cargo test --test sqllogictests --feature avro` or `INCLUDE_TPCH=true cargo test --test sqllogictests --all-features`  would works?

**Comment by duongcongtoai** [comment]
Hi @my-vegetable-has-exploded , could you help me review these small changes, conflict is most likely to happen but i know how to deal with it now ;)

**Comment by my-vegetable-has-exploded** [comment]
> Hi @my-vegetable-has-exploded , could you help me review these small changes, conflict is most likely to happen but i know how to deal with it now ;)

I am not sure the specifics of the issue at hand. For branch conflicts, I think it is okay to `merge` or `rebase` it regenerate the result.

Thanks for your work. @duongcongtoai

**Comment by duongcongtoai** [comment]
> I am not sure the specifics of the problem at hand. For branch conflicts, I think it is okay to merge or rebase it and regenerate the sqllogicaltests.

The current generated output matches main now, i think it is good to review

**Review by my-vegetable-has-exploded — APPROVED** [review]

File: datafusion/sqllogictest/src/engines/datafusion_engine/normalize.rs [hunk 1]
```diff
@@ -96,14 +96,18 @@ fn expand_row(mut row: Vec<String>) -> impl Iterator<Item = Vec<String>> {
...
         let new_lines: Vec<_> = lines
             .into_iter()
-            .map(|l| {
+            .enumerate()
+            .map(|(idx, l)| {
                 // replace any leading spaces with '-' as
                 // `sqllogictest` ignores whitespace differences
                 //
                 // See https://github.com/apache/arrow-datafusion/issues/6328
                 let content = l.trim_start();
```
Comment (alamb) [inline datafusion/sqllogictest/src/engines/datafusion_engine/normalize.rs:105]: I wonder if we can remove the `----` stuff now

Maybe we could remove the `---` and see if the test diff look better 🤔  

Not in this PR

**Review by alamb — APPROVED** [review]
Thanks @duongcongtoai  and @my-vegetable-has-exploded  -- this looks awesome.

We may be able to make the diffs even easier to  check  if we can remove the `---`

**Comment by duongcongtoai** [comment]
Hi @alamb look like only you can merge it :D

**Comment by alamb** [comment]
Thanks again @duongcongtoai  and @my-vegetable-has-exploded  -- I merged up this PR from main and re-ran CI tests just to be sure we didn't introduce any logical conflicts and it all looks good to me. Thanks agian!

**Pull request merged**